File: Aditya Vardhan Singh/Challenge Solutions/Day 5/Python/Position.py
def find_position(arr1, arr2, k):
    arr3 = []
    n = len(arr1)
    m = len(arr2)
    i = j = 0

    while i < n and j < m:
        if arr1[i] < arr2[j]:
            arr3.append(arr1[i])
            i += 1
        else:
            arr3.append(arr2[j])
            j += 1
    
    while i < n:
        arr3.append(arr1[i])
        i += 1
    
    while j < m:
        arr3.append(arr2[j])
        j += 1
    
    return arr3[k-1]


arr1 = [2, 3, 6, 7, 9]
arr2 = [1, 4, 8, 10]
k = 5

print( find_position(arr1, arr2, k) )

File: Aditya Vardhan Singh/Challenge Solutions/Day 5/C++/Position.cpp
#include <iostream>
#include <vector>

using namespace std;

int find_position(vector<int>& arr1, vector<int>& arr2, int k) {
    vector<int> arr3;
    int n = arr1.size();
    int m = arr2.size();
    int i = 0, j = 0;

    while (i < n && j < m) {
        if (arr1[i] < arr2[j]) {
            arr3.push_back(arr1[i]);
            i++;
        } else {
            arr3.push_back(arr2[j]);
            j++;
        }
    }

    while (i < n) {
        arr3.push_back(arr1[i]);
        i++;
    }

    while (j < m) {
        arr3.push_back(arr2[j]);
        j++;
    }

    return arr3[k-1];
}

int main(int argc, const char ** argv) {
    if (argc != 2) {
        printf("Usage: %s <input_file>\n", argv[0]);
        exit(EXIT_FAILURE);
    }

    FILE *fp = fopen(argv[1], "r");
    int data, n;
    vector<int> arr1, arr2;
    int k;

    fscanf(fp, "%d", &n);
    while (n--) {
        fscanf(fp, "%d", &data);
        arr1.push_back(data);
    }

    fscanf(fp, "%d", &n);
    while (n--) {
        fscanf(fp, "%d", &data);
        arr2.push_back(data);
    }

    fscanf(fp, "%d", &k);

    cout << find_position(arr1, arr2, k) << endl;
    return EXIT_SUCCESS;
}